### ISSUE TITLE
bcftools: command does not work correctly. Can you help me?

### ISSUE THREAD

**Issue by jhownatan1** [samtools/bcftools#2318]
Hello, how are you?

So, I used the command `bcftools view -v indels SRZ189891_722g.990.SNP.INDEL.chrAll.vcf.gz -Oz -o indels.vcf.gz` to create a file with only the indels. When I checked if the number of indels in the file SRZ189891_722g.990.SNP.INDEL.chrAll.vcf.gz was the same as in indels.vcf.gz, the results matched. However, I wanted to verify if any SNPs had been filtered out as well, so I used the command `bcftools view -v snps indels.vcf.gz|grep -v -c '^#'` . The result was that there were SNPs in the file indels.vcf.gz, when there should only be indels.

What could have happened? What can I do to have only the indels?

**Comment by jmarshall** [comment]
FYI this was cross-posted to [biostars](https://www.biostars.org/p/9605443/) and [bioinf.SE](https://bioinformatics.stackexchange.com/questions/23018/bcftools-command-does-not-work-correctly-can-you-help-me), which are probably more appropriate venues for this sort of question. (Also: **don't cross-post**.)

**Comment by jhownatan1** [comment]
@pd3 Thanks for helping me out!

So, I used the command `bcftools norm -m -any input.vcf.gz -Oz | bcftools view -v indels -Oz > indels.vcf.gz` and it gave me a file. When I used the command `bcftools view -v snps indels.vcf.gz|grep -v -c '^#'` to see how many snps there were, there were none. However, when I used the command `bcftools view -v indels indelss.vcf.gz|grep -v -c '^#'` to see if it had filtered out the indels correctly, it gave me a value of 26067197, when there were originally 18722319. What could have happened?
